datastore/src/main/resources/META-INF/persistence.xml

```diff
@@ -37,6 +38,43 @@
           <property name="org.hibernate.envers.do_not_audit_optimistic_locking_field" value="false"/>
           <property name="hibernate.generate_statistics" value="true" />
 
+          <!-- Configure batches to improve performance when doing bulk inserts and updates.
+               Do not exceed with the size because when an entity is persisted, Hibernate stores it in the persistence context (in memory) before flushing.
+          -->
+          <property name="hibernate.jdbc.batch_size" value="50"/>
+          <property name="hibernate.order_inserts" value="true"/>
+          <property name="hibernate.order_updates" value="true"/>
+          <property name="hibernate.batch_versioned_data" value="true"/>
+
+          <!-- Raise error instead of warning if “HHH000104: firstResult/maxResults specified with collection fetch; applying in memory!” is detected (since Hibernate 5.2.13) -->
+          <property name="hibernate.query.fail_on_pagination_over_collection_fetch" value="true"/>
+
+          <!--
+              Infinispan second level cache (default Wildfly settings are
+              https://infinispan.org/docs/stable/titles/integrating/integrating.html#default_local_configuration_second_level):
+                - Eviction wake up interval is 5 seconds.
+                - Max number of entries are 10,000.
```

Comment:
It would be good to monitor the size of the cache. My feeling is that higher number might be better for us as we have milions of artifacts and my feeling is that we might fill the cache quite quickly. I'd consider adding a metrics for it and increasing the cache for artifacts.

Reply:
Yes, that is the purpose of the REST endpoints implemented in https://projects.engineering.redhat.com/browse/NCL-5408. They will expose (once 2LC is enabled) all those information, along with the total size of the cache in memory. I have also progressed in displaying many of the in Grafana (https://projects.engineering.redhat.com/browse/NCL-5417), by enhancing the pnc-status service (https://projects.engineering.redhat.com/browse/NCL-5411), so this is all working. The last dashboard will display the stats for the 2nd level cache as per above (rest endpoints now returns no value as 2nd level cache is not enable yet).

Reply:
@vibe13 Great! :-)